### ISSUE TITLE
Add a wait function to the phaster demo notebook

### ISSUE THREAD

**Issue by mbonsma** [phageParser/phageParser#278]
[Phaster](http://phaster.ca/), an outside program that we're using to find prophages in genomes, is occasionally giving a weird error where the API request `requests.get(apiurl)` returns `<Response [520]>`, then a few seconds to minutes later it returns `<Response [200]>`, the normal and correct response. This is probably an issue with Phaster and not us, but to get around it we could add a function that waits and tries again every few seconds until an intolerably long time has passed (maybe 20 seconds?).  

This function would go in the following code block:

```
except ValueError:  # includes simplejson.decoder.JSONDecodeError
    # wait a delay and try again
    result = wait_for_phaster(max_wait)  # THIS IS THE NEW FUNCTION
    # if end result is success, keep going. Otherwise, print an error message
        print("Error acquiring results for accession %s" %acc)
```

An exponentially distributed set of wait times would be great, something like `[1,2,4,7,20]` (or just use an actual exponential distribution).

**Comment by yuzhoux** [comment]
hi @mbonsma , I see you use 'time.sleep(2)' in the third code cell at here:  phageParser/demos/Phaster.ipynb  Is this demo Juypter notebook the place you want to have "exponentially auto retry"?

**Comment by mbonsma** [comment]
@yuzhoux, yes, exactly! The next block also does the same thing and it should be used there too.
